homeassistant/components/nasweb/switch.py

```diff
@@ -0,0 +1,114 @@
+    nasweb_data: NASwebData = hass.data[DOMAIN]
+    coordinator = nasweb_data.entries_coordinators[config.entry_id]
+    nasweb_outputs = coordinator.data[KEY_OUTPUTS]
+    coordinator.async_add_switch_callback = async_add_entities
+    entities: list[RelaySwitch] = []
+    for out in nasweb_outputs:
+        if not isinstance(out, NASwebOutput):
+            _LOGGER.error("Cannot create RelaySwitch entity without NASwebOutput")
+            continue
+        new_output = RelaySwitch(coordinator, out)
+        entities.append(new_output)
+    async_add_entities(entities)
+
+
+class RelaySwitch(SwitchEntity):
+    """Entity representing NASweb Output."""
+
+    def __init__(
+        self, coordinator: DataUpdateCoordinator, nasweb_output: NASwebOutput
+    ) -> None:
+        """Initialize RelaySwitch."""
+        self.coordinator = coordinator
+        self._output = nasweb_output
+        self._attr_is_on = self._output.state
+        self._attr_available = True  # self._output.available
+        self._attr_icon = "mdi:export"
+        self._attr_has_entity_name = True
+        self._attr_should_poll = False
+        self._attr_translation_key = OUTPUT_TRANSLATION_KEY
+        self._attr_unique_id = (
+            f"{DOMAIN}.{self._output.webio_serial}.relay_switch.{self._output.index}"
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Add coordinator update listener when entity is added to hass."""
+        await super().async_added_to_hass()
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self._handle_coordinator_update, None)
+        )
+        self._handle_coordinator_update()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        old_available = self.available
+        self._attr_is_on = self._output.state
+        if (
+            time.time() - self._output.last_update >= STATUS_UPDATE_MAX_TIME_INTERVAL
+            or not self.coordinator.last_update_success
+        ):
+            self._attr_available = False
+        else:
+            self._attr_available = (
+                self._output.available if self._output.available is not None else False
+            )
+        if old_available and self._output.available is None and self.unique_id:
+            _LOGGER.warning("Removing entity: %s", self)
+            er: EntityRegistry = async_get(self.hass)
+            er.async_remove(self.entity_id)
+            return
+        self.async_write_ha_state()
+
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Return DeviceInfo linking this RelaySwitch with NASweb device."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._output.webio_serial)},
+        )
+
+    @property
+    def name(self) -> str:
+        """Return name of RelaySwitch."""
+        translated_name = super().name
+        return f"{translated_name} {self._output.index:2d}"
```

Comment:
Can this be set in `__init__`?

Reply:
Not sure if it's possible. Tried to assign `_attr_name` in `__init__` but `super().name` throws exception related to `self.platform` which seems to not be set at this stage. It is possible to move this into `async_added_to_hass` but then I need to access EntityRegistry and call `async_update_entity` in order to refresh name displayed by frontend. Alternatively, within `name()` i could check if `_attr_name` is set and perform the assignment only when it's `None`.

Reply:
Right, I see. I'd like a second opinion on if it's OK to compose the entity name like this. It seems OK to me, but no other integration does it until now AFAICT.

Reply:
Please change so the NASweb integration has its own translation string `"Switch {index}"` and set the index in the translation_placeholders:
```py
_attr_translation_placeholders = {"index": f"{nasweb_output.index:2d}"}
```